### PR TITLE
Adds URL validation on Place

### PR DESCRIPTION
**Pull request by edward**
Adds URL validation to `Place` to prevent XSS exploits when entering malicious data in the URL field.

@clayton-shopify for review

### CODE REVIEW

**Comment by edward** [comment]
Oops. Didn’t run the full test suite. I’ll fix that up.

**Comment by clayton-shopify** [comment]
Looks good, you just need to put valid URLs in `test/fixtures/places.yml`.

**Comment by edward** [comment]
@clayton-shopify it’s a little more complicated than that since URLs are optional. I’m going to take a look at how Rails built-in validations work to that effect to see how they handle the optionality.

**Comment by clayton-shopify** [comment]
Maybe `allow_blank: true`?

**Comment by edward** [comment]
@clayton-shopify yep! That’s pretty close to what I ended up implementing. TIL how custom validators work in Rails 4

**Comment by clayton-shopify** [comment]
I think you do want `allow_blank` instead of `allow_nil`. I just tested in the browser, and it won't take an empty URL.

File: lib/url_validator.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'uri'
+
+class UrlValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return true if options[:allow_nil] && value.nil?
```
Comment (clayton-shopify) [inline lib/url_validator.rb]: Also, this line appears to be unnecessary.

**Comment by edward** [comment]
Nuts. Sounds like I’m missing a test to simulate the regular form submission. I’ll add an `allow_blank`.

**Comment by clayton-shopify** [comment]
I opened https://github.com/Shopify/forage/pull/11 in case you want to start there.

File: lib/url_validator.rb (hunk 1 above)
Comment (edward) [inline lib/url_validator.rb]: Yep, you’re totally right. Looks like it comes in for free here: https://github.com/rails/rails/blob/3e36db4406beea32772b1db1e9a16cc1e8aea14c/activemodel/lib/active_model/validator.rb#L150

**Pull request merged**